joomla/joomla-cms
# [4.0] Remove com_csp and move the CSP configuration to the HttpHeaders plugin

### Summary of Changes

After discussions within Production it has been decided today (official note will be included in the upcomming meeting minutes) to drop com_csp from Joomla 4.0 and move the "manuall" CSP settings back to the Plugin. While it could be re-implemented in a future version.
This PR now does that by removing the backend and frontend code of com_csp as well as install and update sql logic stuff.

With that removal the collection and autogeneration of CSP rules will be gone from Joomla 4.0 but the Plugin will still allow to setup the CSP rules.

### Testing Instructions

#### Test Case #1 
- Install the build (https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33550/downloads/43088/Joomla_4.0.0-beta8-dev+pr.33550-Development-Full_Package.zip)
- Make sure there is no com_csp any more.
- Confirm that the Plugin System HttpHeaders now includes also the CSP settings.

#### Test Case #2
- Install a 4.0.0beta7
- confirm that com_csp is there
- set the update server to (https://ci.joomla.org/artifacts/joomla/joomla-cms/4.0-dev/33550/downloads/43088/pr_list.xml)
- Update to the new build thats provided
- Confirm that com_csp is gone
- Confirm that the plugin has the new options

### Actual result BEFORE applying this Pull Request

com_csp is there

### Expected result AFTER applying this Pull Request

com_csp is gone

### Documentation Changes Required

- [x] The docs pages for the component and the options page needs to be removed
- [x] The existing doc page needs to be adjusted and updated with the changes done here.

### B/C implications

With the removal of com_csp the auto generated csp rules are gone and they will not be migrated, please extract them before you install the update that drops this feature and set it as Forced HTTP Header or via the new setting implmented here.

### Personal note

I would like to thank all the people that helped to get to this feature to this place specificly @yvesh and @SniperSister who worked together with me to bring that idea to live. As well as all the other people who put work and effort into extending and improving the current implementation until now. I personally still think this is an important feature to the CMS but I will follow the decision taken by Production.
